diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -183,28 +189,59 @@ controller_interface::return_type DiffDriveController::update(
   }
   else
   {
-    double left_position_mean = 0.0;
-    double right_position_mean = 0.0;
-    for (size_t index = 0; index < wheels.wheels_per_side; ++index)
+    if (odom_params_.position_feedback)
     {
-      const double left_position = registered_left_wheel_handles_[index].position.get().get_value();
-      const double right_position =
-        registered_right_wheel_handles_[index].position.get().get_value();
-
-      if (std::isnan(left_position) || std::isnan(right_position))
+      double left_position_mean = 0.0;
+      double right_position_mean = 0.0;
+      for (size_t index = 0; index < wheels.wheels_per_side; ++index)
       {
-        RCLCPP_ERROR(
-          logger, "Either the left or right wheel position is invalid for index [%zu]", index);
-        return controller_interface::return_type::ERROR;
+        const double left_position =
```

Comment:
The code in these two branches of the `if` are essentially the same. Could you please try to do code validation, etc before the `if`?

You could even generalize the print by using the `feedback_type()` function or something alike